wgpu-hal/src/dx12/dcomp.rs

```diff
@@ -1,4 +1,68 @@
+use alloc::sync::Arc;
+use core::{ffi, ptr};
+
+use once_cell::sync::Lazy;
 use windows::Win32::{Foundation::HWND, Graphics::DirectComposition};
+use windows_core::Interface;
+
+use super::DynLib;
+
+// Lazy-loaded DirectComposition library
+#[derive(Debug)]
+pub(crate) struct DCompLib {
+    lib: Lazy<Result<DynLib, crate::SurfaceError>>,
+}
+
+impl DCompLib {
+    pub(crate) fn new() -> Self {
+        Self {
+            lib: Lazy::new(|| {
+                unsafe {
+                    DynLib::new("dcomp.dll").map_err(|err| {
+                        // Stringify because `libloading::Error` is not `Clone`
```

Comment:
This comment is stale, SurfaceError doesn't hold any dynamic information at all.